smtppipeline.go

```diff
@@ -19,6 +19,30 @@ type SMTPPipelineStep interface {
 	Pass(ctx *module.DeliveryContext, msg io.Reader) (io.Reader, bool, error)
 }
 
+type verifySourceHostnameStep struct{}
+
+func (verifySourceHostnameStep) Pass(ctx *module.DeliveryContext, _ io.Reader) (io.Reader, bool, error) {
+	ipAddr, ok := ctx.SrcAddr.(*net.IPAddr)
+	if !ok {
+		return nil, true, nil
+	}
+
+	srcIPs, err := net.LookupIP(ctx.SrcHostname)
+	if err != nil {
+		ctx.Ctx["src_hostname_check"] = false
+		return nil, false, errors.New("Source hostname does not resolve")
```

Comment:
Nitpick: [Error messages should not be capitalized](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)